src/Core/Field.h

```diff
@@ -580,6 +592,11 @@ class Field
             case Types::Array:   return f(field.template get<Array>());
             case Types::Tuple:   return f(field.template get<Tuple>());
             case Types::Map:     return f(field.template get<Map>());
+            case Types::Bool:
+            {
+                bool value = bool(field.template get<UInt64>());
+                return f(value);
+            }
```

Comment:
I'd write single line as in other cases, but actually up to you.
```suggestion
            case Types::Bool: return f(static_cast<bool>(field.template get<UInt64>()));
```

Reply:
Couldn't write it like that, because there are cases when `f` accepts value by non const reference. But I think It would be better to change that places to const reference. Will do it

Reply:
UPD: can't change all places to const reference, because some of them change the value (for example FieldVisitorSum)